src/contracts/GPv2Settlement.sol

```diff
@@ -13,7 +14,14 @@ contract GPv2Settlement {
     /// GPv2 contracts.
     bytes32 internal immutable domainSeparator;
 
-    constructor() {
+    /// @dev The authenticator is used to determine who can call the settle function.
+    /// That is, only authorised solvers have the ability to invoke settlements.
+    /// Any valid authenticator implements an isSolver method called by the onlySolver
+    /// modifier below.
+    GPv2Authentication private immutable authenticator;
+
+    constructor(GPv2Authentication _authenticator) {
```

Comment:
nit: We should follow the solidity style guide: https://docs.soliditylang.org/en/v0.7.5/style-guide.html#avoiding-naming-collisions

```suggestion
    constructor(GPv2Authentication authenticator_) {
```

Reply:
Fine.